x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -672,6 +677,33 @@ public List<Alias> visitFields(EsqlBaseParser.FieldsContext ctx) {
         return ctx != null ? visitList(this, ctx.field(), Alias.class) : new ArrayList<>();
     }
 
+    @Override
+    public NamedExpression visitAggField(EsqlBaseParser.AggFieldContext ctx) {
+        Source source = source(ctx);
+        Alias field = visitField(ctx.field(), source);
+        var filterExpression = ctx.booleanExpression();
+        Expression condition = filterExpression != null ? expression(filterExpression) : null;
+
+        if (condition != null) {
```

Comment:
Not following. expression is called only if the context is not null so it should return null.

Reply:
It's just a nit: `filterExpression` and `condition` have the same nullability, since `expression()` will never return `null`: if  `filterExpression` is not-null, `condition` will be not-null. So no need to check twice against `null`.

Reply:
ahhhh - now I see it!
Thanks for bearing with me.